### ISSUE TITLE
ignoreChanges with class Data_Group is ignored when URL contains @ 

### ISSUE THREAD

**Issue by megamattzilla** [F5Networks/f5-appsvcs-extension#569]
### Environment
 * Application Services Version: 3.30 and 3.34
 * BIG-IP Version: 15.1.5

### Summary
When data group is defined with a URL containing an @ (example: when using github private repo token) ignoreChanges does not behave as expected. 

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "class": "AS3",
    "action": "deploy",
    "persist": true,
    "declaration": {
      "class": "ADC",
      "schemaVersion": "3.30.0",
      "id": "123abc",
      "Common": {
        "class": "Tenant",
        "Shared": {
          "class": "Application",
          "template": "shared",
        "test-datagroup": {
          "class": "Data_Group",
          "storageType": "external",
          "keyDataType": "string",
          "externalFilePath": "https://test@raw.githubusercontent.com/megamattzilla/f5PublicConfigCommon/main/dataGroups/header_to_pool.txt",
          "ignoreChanges": true,
          "separator": ":=",
          "label": "test-datagroup"
        }
          }
        }
      }
    }

```
2. Modify the remote file `header_to_pool.txt` with a change. 

3.  Resend exact declaration from step 1. 

4. Observe the following response:
```json
    "results": [
        {
            "code": 200,
            "message": "success",
            "lineCount": 16,
            "host": "localhost",
            "tenant": "Common",
            "runTime": 3242
        },
        {
            "code": 200,
            "message": "success",
            "lineCount": 19,
            "host": "localhost",
            "tenant": "Common",
            "runTime": 3124
        }
    ],
```

### Expected Behavior. 

Expected AS3 response:
```json
        {
            "code": 200,
            "message": "no change",
            "host": "localhost",
            "tenant": "Common",
            "runTime": 2012
        },
        {
            "code": 200,
            "message": "no change",
            "host": "localhost",
            "tenant": "Common",
            "runTime": 2049
        }
```
The Big-IP data group should NOT be updated with the remote file contents when ignoreChanges is set to true.  
 This should not happen according to the schema [here](https://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/latest/refguide/schema-reference.html#data-group).

### Actual Behavior
The Big-IP data group is updated with the remote file contents when ignoreChanges set to true.

**Comment by megamattzilla** [comment]
Strangely, I've only been able to reproduce when the URL contains a `@`. Without the `@` the second POST reflects `"no change"` as expected.

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have added this issue to our internal product backlog as AUTOTOOL-3041.

**Comment by dstokesf5** [comment]
This was resolved in the 3.36.0 release of AS3.
